besu/src/main/java/org/hyperledger/besu/controller/BesuController.java

```diff
@@ -191,6 +192,21 @@ BesuControllerBuilder fromGenesisConfig(
           genesisConfig.getConfigOptions(genesisConfigOverrides);
       final BesuControllerBuilder builder;
 
+      // TODO-lucas Too much in-line code?
+      if (configOptions.isQbft() && (configOptions.isIbft2() || configOptions.isIbftLegacy())) {
+        final BesuControllerBuilder originalControllerBuilder;
+        if (configOptions.isIbft2()) {
+          originalControllerBuilder = new IbftBesuControllerBuilder();
+        } else if (configOptions.isIbftLegacy()) {
+          originalControllerBuilder = new IbftLegacyBesuControllerBuilder();
+        } else {
+          throw new IllegalStateException("Invalid config");
+        }
```

Comment:
This is duplicating a lot of the below code. Perhaps which builder to use could be just moved into it's own function.